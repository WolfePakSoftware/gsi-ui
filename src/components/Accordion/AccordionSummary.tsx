import React, { Component } from 'react';
import {
  Accordion,
  AccordionTitleProps,
  Icon,
  Segment,
  SemanticICONS
} from 'semantic-ui-react';
import { accordionStyle, TitleM, titleIcon, titleStyle } from './style';

interface IAccordionState {
  activeIndex: any;
  icon: SemanticICONS;
}

interface IAccordionProps {
  title: string;
  titleBackgroundColor?: string;
  titleBorderColor?: string;
  iconActive: SemanticICONS;
  iconDeactivate: SemanticICONS;
  className?: string;
}

export class AccordionSummary extends Component<
  IAccordionProps,
  IAccordionState
> {
  constructor(props: IAccordionProps) {
    super(props);
    this.state = {
      activeIndex: 0,
      icon: this.props.iconDeactivate
    };
  }

  handleClick = (
    e: React.MouseEvent<HTMLDivElement>,
    titleProps: AccordionTitleProps
  ) => {
    e.preventDefault();
    const { index } = titleProps;
    const { activeIndex } = this.state;
    const { iconActive, iconDeactivate } = this.props;
    const newIndex = index === activeIndex ? -1 : index;
    const icon = index === activeIndex ? iconActive : iconDeactivate;
    this.setState({ icon, activeIndex: newIndex });
  };

  render() {
    const {
      title,
      titleBackgroundColor,
      titleBorderColor,
      className
    } = this.props;
    const { activeIndex, icon } = this.state;

    return (
      <Accordion styled style={accordionStyle} className={className}>
        <TitleM
          background={titleBackgroundColor}
          border={titleBorderColor}
          active={activeIndex === 0}
          index={0}
          onClick={this.handleClick}
        >
          <Icon style={titleIcon} name={icon} />
          <span>{title}</span>
        </TitleM>
        <Accordion.Content active={activeIndex === 0}>
          {this.props.children}
        </Accordion.Content>
      </Accordion>
    );
  }
}
