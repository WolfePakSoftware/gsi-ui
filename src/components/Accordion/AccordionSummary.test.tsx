import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { AccordionSummary } from './AccordionSummary';
import { Segment } from 'semantic-ui-react';
import React from 'react';

it('renders without crashing Accordion Summary less properties', () => {
  const div = document.createElement('div');
  const amount = 5;
  ReactDOM.render(
    <AccordionSummary
      iconActive={'plus' as 'plus'}
      iconDeactivate={'minus' as 'minus'}
      title={`Delivery Summary (${amount})`}
    >
      <Segment>
        <p>Hi world it's me from here!!!!!!!!!!</p>
      </Segment>
    </AccordionSummary>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing Accordion Summary All properties', () => {
  const div = document.createElement('div');
  const amount = 5;
  ReactDOM.render(
    <AccordionSummary
      iconActive={'plus' as 'plus'}
      iconDeactivate={'minus' as 'minus'}
      title={`Delivery Summary (${amount})`}
      titleBackgroundColor={'#cbe8ba'}
      titleBorderColor={'#4c7a34'}
    >
      <Segment>
        <p>Hi world it's me from here!!!!!!!!!!</p>
      </Segment>
    </AccordionSummary>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('render correctly Accordion All properties', () => {
  const amount = 5;
  const AccordionTest = renderer
    .create(
      <AccordionSummary
        iconActive={'plus' as 'plus'}
        iconDeactivate={'minus' as 'minus'}
        title={`Delivery Summary (${amount})`}
        titleBackgroundColor={'#cbe8ba'}
        titleBorderColor={'#4c7a34'}
      >
        <Segment>
          <p>Hi world it's me from here!!!!!!!!!!</p>
        </Segment>
      </AccordionSummary>
    )
    .toJSON();
  expect(AccordionTest).toMatchSnapshot();
});

it('render correctly Accordion less properties', () => {
  const amount = 5;
  const AccordionTest = renderer
    .create(
      <AccordionSummary
        iconActive={'plus' as 'plus'}
        iconDeactivate={'minus' as 'minus'}
        title={`Delivery Summary (${amount})`}
      >
        <Segment>
          <p>Hi world it's me from here!!!!!!!!!!</p>
        </Segment>
      </AccordionSummary>
    )
    .toJSON();
  expect(AccordionTest).toMatchSnapshot();
});
