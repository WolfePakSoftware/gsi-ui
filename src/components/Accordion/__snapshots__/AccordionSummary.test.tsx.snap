// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Accordion All properties 1`] = `
<div
  className="accordion ui styled"
  style={
    Object {
      "marginBottom": "25px",
      "width": "100%",
    }
  }
>
  <div
    className="active title sc-bdVaJa hnjgTQ"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="minus icon"
      onClick={[Function]}
      style={
        Object {
          "float": "left",
        }
      }
    />
    <span>
      Delivery Summary (5)
    </span>
  </div>
  <div
    className="content active"
  >
    <div
      className="ui segment"
    >
      <p>
        Hi world it's me from here!!!!!!!!!!
      </p>
    </div>
  </div>
</div>
`;

exports[`render correctly Accordion less properties 1`] = `
<div
  className="accordion ui styled"
  style={
    Object {
      "marginBottom": "25px",
      "width": "100%",
    }
  }
>
  <div
    className="active title sc-bdVaJa hBYAyW"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="minus icon"
      onClick={[Function]}
      style={
        Object {
          "float": "left",
        }
      }
    />
    <span>
      Delivery Summary (5)
    </span>
  </div>
  <div
    className="content active"
  >
    <div
      className="ui segment"
    >
      <p>
        Hi world it's me from here!!!!!!!!!!
      </p>
    </div>
  </div>
</div>
`;
